- Description: lcs: net_device reference counting problem
  Symptom:     Trying to do an rmmod lcs or ungrouping an lcs device results
               in a hanging session with messages "waiting for ... to
               become free" displayed.
  Problem:     An in_dev_put was missing in multicast address list handling.
  Solution:    Added the missing in_dev_put.
  Problem-ID:  --
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/net/lcs.c	11 May 2004 08:59:46 -0000	1.74.2.1
+++ linux-2.5/drivers/s390/net/lcs.c	14 May 2004 07:30:37 -0000	1.74.2.2
@@ -11,7 +11,7 @@
  *			  Frank Pavlic (pavlic@de.ibm.com) and
  *		 	  Martin Schwidefsky <schwidefsky@de.ibm.com>
  *
- *    $Revision: 1.74.2.1 $	 $Date: 2004/05/11 08:59:46 $
+ *    $Revision: 1.74.2.2 $	 $Date: 2004/05/14 07:30:37 $
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -58,7 +58,7 @@
 /**
  * initialization string for output
  */
-#define VERSION_LCS_C  "$Revision: 1.74.2.1 $"
+#define VERSION_LCS_C  "$Revision: 1.74.2.2 $"
 
 static char version[] __initdata = "LCS driver ("VERSION_LCS_C "/" VERSION_LCS_H ")";
 static char debug_buffer[255];
@@ -1047,6 +1047,7 @@
 	spin_unlock(&card->ipm_lock);
 	read_unlock(&in4_dev->lock);
 	lcs_fix_multicast_list(card);
+	in_dev_put(in4_dev);
 	return 0;
 }
 /**

